src/payload.rs

```diff
@@ -612,7 +614,10 @@ pub extern "C" fn z_bytes_iterator_next(
 ) -> bool {
     match iter.transmute_mut().next() {
         Some(buf) => {
-            Inplace::init(out.transmute_uninit_ptr(), buf.into());
+            // this is safe because anything is convertible to ZBytes
+            Inplace::init(out.transmute_uninit_ptr(), unsafe {
```

Comment:
Just had talk with @Mallets about that and he pointed out that this is fallible in fact. 
This line https://github.com/eclipse-zenoh/zenoh/blob/dev/1.0.0/zenoh/src/api/bytes.rs#L227 reads length and buffer itself. If length for some reason is larger than remaining buffer, `DidntRead` error is returned